core/src/main/scala/chisel3/internal/BiConnect.scala

```diff
@@ -179,9 +179,10 @@ private[chisel3] object BiConnect {
           )
         ) {
           pushCommand(Connect(sourceInfo, leftReified.get.lref, rightReified.get.lref))
-        } else if (!emitStrictConnects) {
-          newLeft.legacyConnect(newRight)(sourceInfo)
```

Comment:
I'm confused as to why you are not emitting it, but rather correctly determining the `newLeft`/`newRight` ?

Reply:
how should i correctly determine the newLeft/newRight? And how different will it be from just using the algorithms we already have?

Reply:
Took your suggestion, #2801 is the version of this that just fixes the canBeSink/canBeSource logic and keeps the legacy connects here.